JoinColony/colonyDapp
# Motions and Disputes Create Subdomain Motion and Escalate It

## Description

This PR adds a domains dropdown on the action modals, when in motion mode, and adds the ability to create a motion in a subdomain and vote on in in that subdomain. Upon finalizing the vote, you'll also have the ability to escalate it, if you think the result of the vote was not satisfactory

The modals that will have the full domains dropdown:
- Payment
- Edit Team
- Permissions

The other ones will still show the domains dropdown, but with it in a disabled state, due to only being possible to create the specific motion in the root domain

**Note #1** Escalation has been disabled for the current released due to it being complicated and too time consuming to make sense now. This was decided alongside product.

**To test**
- Create subdomain
- Give yourself reputation in that subdomain (create a payment from that subdomain)
- Create one of the supported motion types: Payment, Edit Team or Permissions but make sure to create it in the subdomain 
- Stake both sides
- Vote
- You can now escalate it

**Changes** 

- [x] Refactored `ColonyDomainSelector` into `ColonyHome`
- [x] Extracted `DomainDropdown` as a `core` component
- [x] Create dashboard `MotionDomainSelect` component
- [x] Add motion subdomain vote logic to `TokenMintDialog`
- [x] Add motion subdomain vote logic to `NetworkContractUpdateDialog`
- [x] Add motion subdomain vote logic to `CreateDomainDialog`
- [x] Add motion subdomain vote logic to `TransferFundsDialog`
- [x] Add motion subdomain vote logic to `TokenEditDialog`
- [x] Add motion subdomain vote logic to `EditColonyDetailsDialog`
- [x] Add motion subdomain vote logic to `TokenMintDialog`
- [x] Add motion subdomain vote logic for all domains to `CreatePaymentDialog`
- [x] Update `createPaymentMotion` saga with the motion vote domain
- [x] Add motion subdomain vote logic for all domains to `EditDomainDialog`
- [x] Update `createEditDomainMotion` saga with the motion vote domain
- [x] Add motion subdomain vote logic for all domains to `PermissionManagementDilaog`
- [x] Update `managePermissionsMotion` saga with the motion vote domain
- [x] Add `Tag` component `dangerGhost` theme
- [x] Add escalation to `MOTION_TAG_MAP`
- [x] Add escalation system message
- [x] Add `escalateMotion` saga
- [x] Add escalation controls (and display logic) to the `DefaultMotion` component

**Other fixes**

- [x] Fixed core `Select` component to have a fallback for the active selected item
- [x] Fixed core `Toggle` can now handle onChange callbacks
- [x] Refactore `PermissionManagementDilaog` code structure
- [x] Fixed `DefaultMotion` correct _threshold_ naming

**Demo**

![Screenshot from 2021-05-27 00-59-56](https://user-images.githubusercontent.com/1193222/119737800-71435580-be88-11eb-98f5-8a6e06b6a749.png)

![Screenshot from 2021-05-27 01-00-02](https://user-images.githubusercontent.com/1193222/119737814-71dbec00-be88-11eb-9137-5cb641696964.png)

![Screenshot from 2021-05-27 01-00-13](https://user-images.githubusercontent.com/1193222/119737815-71dbec00-be88-11eb-8be4-9cfcfeeb3946.png)

![Screenshot from 2021-05-27 00-59-28](https://user-images.githubusercontent.com/1193222/119737848-743e4600-be88-11eb-8b2d-58b4be73230d.png)

Note the currently disabled escalate button

![Screenshot from 2021-05-27 21-41-18](https://user-images.githubusercontent.com/1193222/119881610-22ec9000-bf36-11eb-8f64-0145fb0bea14.png)

Resolves DEV-349
Resolves DEV-378
